src/main/java/org/generationcp/middleware/service/impl/user/UserServiceImpl.java

```diff
@@ -201,26 +217,58 @@ public Integer updateUser(final UserDto userDto) {
 			user.setCloseDate(currentDate);
 			user.setStatus(userDto.getStatus());
 
-			// update user roles to the particular user
-			final UserRole role = user.getRoles().get(0);
-			if (!role.getRole().equals(userDto.getRole())) {
-				role.setRole(userDto.getRole());
+			final List<UserRole> userRoles = new ArrayList<>();
+			if (userDto.getUserRoles() != null) {
+				for (final UserRoleDto userRoleDto : userDto.getUserRoles()) {
+					boolean found = false;
+					for (final UserRole userRole : user.getRoles()) {
+						if (userRole.getRole().getId().equals(userRoleDto.getRole().getId()) &&
```

Comment:
This if statetement with sub criteria of && and || is SUPER hard to understand. 
You can extract the relevant sub-criteria to meaning methods.